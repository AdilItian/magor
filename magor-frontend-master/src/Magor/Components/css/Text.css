.font {
    font-family: 'Roboto';
    font-weight: normal;
}

.titleText {
    font-size: 96px;
    color: #1faeff;
    margin: 20px 0;
}

.titleText.small {
    font-size: 32px;
    color: #1faeff;
    margin: 4px 10px;
}

.titleText.thin {
    font-weight: normal;
}

.titleText span {
    margin-right: 7px;
}

.paraText {
    font-size: 18px;
    color: #000;
}

.blockLink {
    border: 1px solid #ccc;
    border-radius: 10;
    width: 75%;
    margin: 10px 0;
    padding: 10px 20px 20px 20px;
    text-decoration: none;
    cursor: pointer;
    color: inherit;
}

.tag {
    box-sizing: border-box;
    padding: 2px 8px;
    margin: 2px;
    border-radius: 10px;
    background: #f1f3f4;
    color: black;
    font-size: 16px;
}

.tag:hover {
    background: #d1d2d4;
}

.tag.highlighted {
    border-radius: 30px;
}

.tag.highlighted:hover {
    background: #ff0;
}

.tag.action {
    cursor: pointer;
}

.tag.showCross {
    padding-right: 25px;
    background-image: url(/close.svg);
    background-size: 13px 13px;
    background-position: calc(100% - 7px) center;
    background-repeat: no-repeat;
}

.titleText .highlighted {
    background: transparent;
    padding: 3px 7px;
    border: 2px solid #1faeff;
}

p .highlighted {
    padding: 2px 7px;
    border-radius: 5px;
}

p span {
    margin-right: 5px;
}

.highlighted {
    display: inline;
    background: #ff0;
    padding: 4px 12px;
    border-radius: 10px;
}