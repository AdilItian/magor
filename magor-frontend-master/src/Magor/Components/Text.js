import React from 'react'
import cx from '../Utils/cx'
import './css/Text.css'
import shouldHighlight from '../Utils/shouldHighlight'

const highlightQuery = (displaySentence, querySentence) => {
    if (!querySentence) return displaySentence
    const displayWords = displaySentence.split(' ')
    const queryWords = querySentence.split(' ')
    const finalWords = []
    let _key = 0;
    for (let displayWord of displayWords) {
        let highlighted
        if (shouldHighlight(displayWord, queryWords)) {
            highlighted = true
        }
        finalWords.push(
            highlighted ? (
                <span className="highlighted" key={_key++}>
                    {displayWord}
                </span>
            ) : (
                `${displayWord} `
            )
        )
    }
    return finalWords
}

export const Title = (props) => {
    const { query, text, small, thin, ...otherProps } = props
    const titleText = true
    return (
        <h1 className={cx({ titleText, small, thin })} {...otherProps}>
            {highlightQuery(text, query)}
        </h1>
    )
}

export const Para = (props) => {
    if (props.query) {
        return (
            <p {...props} className={cx(['font', props.className])}>
                {highlightQuery(props.text, props.query)}
            </p>
        )
    }
    return (
        <p
            {...props}
            className={cx(['font', props.className])}
            style={props.style}
        >
            {props.text || props.children}
        </p>
    )
}

export const BlockLink = (props) => (
    <a
        href={props.href}
        className={cx('blockLink', props.className)}
        style={props.style}
    >
        {props.children}
    </a>
)

export const Tag = ({ name, showCross, queryWords, ...otherProps }) => (
    <span
        {...otherProps}
        className={cx({
            font: true,
            tag: true,
            highlighted: shouldHighlight(name, queryWords),
            showCross: showCross === true,
            action: !!otherProps.onClick,
        })}
    >
        {name}
    </span>
)
