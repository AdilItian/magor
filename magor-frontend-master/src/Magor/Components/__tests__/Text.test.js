import React from 'react'
import { Tag, BlockLink, Para, Title } from '../Text'
import renderer from 'react-test-renderer'

test('All should be defined', () => {
    expect(Tag).toBeDefined()
    expect(BlockLink).toBeDefined()
    expect(Para).toBeDefined()
    expect(Title).toBeDefined()
})

test('Tag snapshot tests', () => {
    const basicTag = renderer.create(<Tag name="tag" />)
    expect(basicTag.toJSON()).toMatchSnapshot()
    const tagWithCross = renderer.create(<Tag name="tag" showCross />)
    expect(tagWithCross.toJSON()).toMatchSnapshot()
    const highlightedTag = renderer.create(
        <Tag name="tag" queryWords={['tag']} />
    )
    expect(highlightedTag.toJSON()).toMatchSnapshot()
})

test('BlockLink snapshot tests', () => {
    const blockLink = renderer.create(
        <BlockLink href="#">
            <div>This is a BlockLink</div>
        </BlockLink>
    )
    expect(blockLink.toJSON()).toMatchSnapshot()

    const blockLinkWithStyle = renderer.create(
        <BlockLink href="#" style={{ background: 'blue' }}>
            <div>This is a BlockLink with blue background</div>
        </BlockLink>
    )
    expect(blockLinkWithStyle.toJSON()).toMatchSnapshot()
})

test('Para snapshot tests', () => {
    const para = renderer.create(
        <Para className="customClass" style={{ fontWeight: 'bold' }}>
            This is a paragraph.
        </Para>
    )
    expect(para.toJSON()).toMatchSnapshot()

    const paraWithQuery = renderer.create(
        <Para
            className="customClass"
            style={{ fontWeight: 'bold' }}
            query="random query"
            text="This is a paragraph."
        />
    )
    expect(paraWithQuery.toJSON()).toMatchSnapshot()

    const paraWithMatchingQuery = renderer.create(
        <Para
            className="customClass"
            style={{ fontWeight: 'bold' }}
            query="mathing query paragraph"
            text="This is a paragraph."
        />
    )
    expect(paraWithMatchingQuery.toJSON()).toMatchSnapshot()
})

test('Title snapshot tests', () => {
    const title = renderer.create(<Title text="this is a title" />)
    expect(title.toJSON()).toMatchSnapshot()
    expect(title.toJSON().props.className).toBe('titleText')

    const titleSmall = renderer.create(<Title small text="this is a title" />)
    expect(titleSmall.toJSON()).toMatchSnapshot()
    expect(titleSmall.toJSON().props.className).toBe('titleText small')

    const titleThin = renderer.create(<Title thin text="this is a title" />)
    expect(titleThin.toJSON()).toMatchSnapshot()
    expect(titleThin.toJSON().props.className).toBe('titleText thin')

    const titleWithQuery = renderer.create(
        <Title query="random query" text="this is a title" />
    )
    expect(titleWithQuery.toJSON()).toMatchSnapshot()
    expect(titleWithQuery.toJSON().props.className).toBe('titleText')

    const titleWithMatchingQuery = renderer.create(
        <Title query="matching title query" text="this is a title" />
    )
    expect(titleWithMatchingQuery.toJSON()).toMatchSnapshot()
    expect(titleWithMatchingQuery.root.findByProps({className: 'highlighted'}).children[0]).toBe('title')
})
