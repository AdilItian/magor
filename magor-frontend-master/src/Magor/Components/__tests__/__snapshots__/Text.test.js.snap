// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockLink snapshot tests 1`] = `
<a
  className=""
  href="#"
>
  <div>
    This is a BlockLink
  </div>
</a>
`;

exports[`BlockLink snapshot tests 2`] = `
<a
  className=""
  href="#"
  style={
    Object {
      "background": "blue",
    }
  }
>
  <div>
    This is a BlockLink with blue background
  </div>
</a>
`;

exports[`Para snapshot tests 1`] = `
<p
  className="font customClass"
  style={
    Object {
      "fontWeight": "bold",
    }
  }
>
  This is a paragraph.
</p>
`;

exports[`Para snapshot tests 2`] = `
<p
  className="font customClass"
  query="random query"
  style={
    Object {
      "fontWeight": "bold",
    }
  }
  text="This is a paragraph."
>
  This 
  is 
  a 
  paragraph. 
</p>
`;

exports[`Para snapshot tests 3`] = `
<p
  className="font customClass"
  query="mathing query paragraph"
  style={
    Object {
      "fontWeight": "bold",
    }
  }
  text="This is a paragraph."
>
  This 
  is 
  a 
  <span
    className="highlighted"
  >
    paragraph.
  </span>
</p>
`;

exports[`Tag snapshot tests 1`] = `
<span
  className="font tag"
>
  tag
</span>
`;

exports[`Tag snapshot tests 2`] = `
<span
  className="font tag showCross"
>
  tag
</span>
`;

exports[`Tag snapshot tests 3`] = `
<span
  className="font tag highlighted"
>
  tag
</span>
`;

exports[`Title snapshot tests 1`] = `
<h1
  className="titleText"
>
  this is a title
</h1>
`;

exports[`Title snapshot tests 2`] = `
<h1
  className="titleText small"
>
  this is a title
</h1>
`;

exports[`Title snapshot tests 3`] = `
<h1
  className="titleText thin"
>
  this is a title
</h1>
`;

exports[`Title snapshot tests 4`] = `
<h1
  className="titleText"
>
  this 
  is 
  a 
  title 
</h1>
`;

exports[`Title snapshot tests 5`] = `
<h1
  className="titleText"
>
  this 
  is 
  a 
  <span
    className="highlighted"
  >
    title
  </span>
</h1>
`;
